android/src/main/java/com/qliktrialreactnativestraighttable/HeaderViewFactory.java

```diff
@@ -52,88 +49,62 @@ public List<DataColumn> getDataColumns() {
     return dataColumns;
   }
 
-  HeaderViewFactory(ReadableArray readableArray, CustomHorizontalScrollView scrollView, FrameLayout rootView, TableView tableView) {
-    this.scrollView = scrollView;
-    this.rootView = rootView;
-    this.tableView = tableView;
-    if (readableArray != null) {
-      for(int i = 0; i < readableArray.size(); i++) {
-        DataColumn column = new DataColumn(readableArray.getMap(i));
-        dataColumns.add(column);
-      }
-    }
-  }
 
   @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
-  public HeaderViewFactory(ReadableArray readableArray, ReadableArray footerArray,  CustomHorizontalScrollView scrollView, FrameLayout rootView, TableView tableView) {
-    Context context = tableView.getContext();
-    this.scrollView = scrollView;
-    this.rootView = rootView;
-    this.tableView = tableView;
-    if(readableArray != null) {
-      buildHeader(readableArray, context);
-    }
-    if(footerArray != null) {
-      buildFooter(footerArray, context);
-    }
-  }
-
-  public HeaderViewFactory(HeaderView headerView, CustomHorizontalScrollView scrollView, TableView tableView) {
-    this.scrollView = scrollView;
-    this.headerView = headerView;
+  public HeaderViewFactory(List<DataColumn> dataColumns, TableView tableView,  Context context) {
     this.tableView = tableView;
+    this.dataColumns = dataColumns;
+    buildHeader(context);
   }
 
-
-  public static HeaderCell buildFixedColumnCell(FrameLayout rootView, DataColumn column, CustomHorizontalScrollView scrollView) {
+  public static HeaderCell buildFixedColumnCell(FrameLayout rootView, DataColumn column, TableView tableView) {
     int padding = (int) PixelUtils.dpToPx(16);
 
-    HeaderCell fixedFirstHeaderCell = new HeaderCell(rootView.getContext(), column, scrollView);
+    HeaderCell fixedFirstHeaderCell = new HeaderCell(rootView.getContext(), column, tableView);
     fixedFirstHeaderCell.setMaxLines(1);
     fixedFirstHeaderCell.setTypeface(fixedFirstHeaderCell.getTypeface(), Typeface.BOLD);
     fixedFirstHeaderCell.setEllipsize(TextUtils.TruncateAt.END);
     fixedFirstHeaderCell.setTextColor(Color.BLACK);
     fixedFirstHeaderCell.setText(column.label);
     fixedFirstHeaderCell.setPadding(padding, 0, padding, 0);
-    LinearLayout.LayoutParams layoutParams = new LinearLayout.LayoutParams(column.width, TableTheme.headerHeight);
+    LinearLayout.LayoutParams layoutParams = new LinearLayout.LayoutParams((int)column.width, TableTheme.headerHeight);
```

Comment:
```suggestion
    LinearLayout.LayoutParams layoutParams = new LinearLayout.LayoutParams((column.width, TableTheme.headerHeight);
```
isn't column.width already an integer? It's cast a few times in this file